packages/monitoring/src.ts/reserve-data.ts

```diff
@@ -0,0 +1,164 @@
+  return "";
+}
+
+const vmexAlertsDiscordWebhook = process.env.DISCORD_ALERTS_WEBHOOK_URL;
+
+export function sendAlert(
+  message: string,
+  severity: number,
+  messages: string[]
+) {
+  let alertMessage = `SEV ${severity} ALERT: ${message}`;
+  if (severity <= 2) {
+    alertMessage += "\n@everyone";
+  }
+  console.error(alertMessage);
+  messages.push(alertMessage);
+  // TODO: More alerting depending on severity
+}
+
+function sendMessage(message: string) {
+  if (!vmexAlertsDiscordWebhook) {
+    throw new Error(
+      "Vmex discord messages webhook is not defined in env variables"
+    );
+  }
+  const data = typeof message === "string" ? { content: message } : message;
+  return new Promise<void>((resolve, reject) => {
+    fetch(vmexAlertsDiscordWebhook, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(data),
+    })
+      .then((response) => {
+        if (!response.ok) {
+          reject(new Error(`Could not send message: ${response.status}`));
+        }
+        resolve();
+      })
+      .catch((error) => {
+        console.error(error);
+        reject(error);
+      });
+  });
+}
+
+function checkPermanentLoss(
+  market: AssetSummary,
+  network: string,
+  messages: string[]
+) {
+  // amount underlying held by aToken >= aToken outstanding supply - debtToken outstanding supply
+  // this means the amount underlying the atoken holds must be greater than amount underlying users
+  // have claim to less amount underlying users owe
+
+  // the amount we need to pay lenders if all borrowers are liquidated
+  let accountsPayable = market.totalSupplied.sub(market.totalBorrowed);
+
+  if (market.totalReserves.add(market.totalStaked).lt(accountsPayable)) {
+    sendAlert(
+      `
+        NETWORK: ${network}
+        Reserve ${market.asset} on tranche ${market.tranche} lost funds.
+        Reserve holds ${market.totalReserves.toString()}
+        Reserve has staked ${market.totalStaked.toString()}
+        aTokenSupply (amount protocol owes lenders): ${market.totalSupplied.toString()}
+        debtTokenSupply (amount borrowers owes protocol): ${market.totalBorrowed.toString()}
+        deficit: ${accountsPayable.sub(market.totalReserves).toString()}
+        `,
+      1,
+      messages
+    );
+  }
+
+  // console.log(
+  //   `tranche ${market.tranche} reserve ${
+  //     market.asset
+  //   } has an unrealized gain of \$${market.totalReserves
+  //     .sub(accountsPayable)
+  //     .div(scalingFactor)
+  //     .toString()}`
+  // );
+  // console.log(
+  //   `it also has ${market.totalSupplied
+  //     .div(scalingFactor)
+  //     .toString()} supplied and ${market.totalBorrowed
+  //     .div(scalingFactor)
+  //     .toString()} borrows`
+  // );
+}
+
+function checkOracle(
+  market: AssetSummary,
+  network: string,
+  messages: string[]
+) {
+  if (market.currentPriceETH.eq("0")) {
+    // oracle for the market is down
+    sendAlert(
```

Comment:
Can make this slightly more descriptive. Check if it's a chainlink oracle, and then check if the last update timestamp is over the heartbeat. If so, can be more descriptive on why it's down. Otherwise, send generic alert below.

Reply:
Lets iterate this later. We would need to change the eth call contracts. The current solution will definetely warn us if somethign is wrong, we just have to do a bit more manual work.